compiler/rustc_windows_rc/rustc.rc.in

```diff
@@ -0,0 +1,40 @@
+// A template for the rustc_driver and rustc Windows resource files.
+// This file is processed by the build script to produce rustc.rc and rustc_driver.rc
+// with the appropriate version information filled in.
+
+// All the strings are in UTF-8
+#pragma code_page(65001)
+
+#define RUSTC_FILEDESCRIPTION_STR "@RUSTC_FILEDESCRIPTION_STR@"
+#define RUSTC_FILETYPE @RUSTC_FILETYPE@
+#define RUSTC_FILEVERSION_QUAD @RUSTC_FILEVERSION_QUAD@
+#define RUSTC_FILEVERSION_STR "@RUSTC_FILEVERSION_STR@"
+
+#define RUSTC_PRODUCTNAME_STR "@RUSTC_PRODUCTNAME_STR@"
+#define RUSTC_PRODUCTVERSION_QUAD @RUSTC_PRODUCTVERSION_QUAD@
+#define RUSTC_PRODUCTVERSION_STR "@RUSTC_PRODUCTVERSION_STR@"
+
+
+1 VERSIONINFO
+FILEVERSION RUSTC_FILEVERSION_QUAD
+PRODUCTVERSION RUSTC_PRODUCTVERSION_QUAD
+FILEOS  0x00040004
```

Comment:
Do we want `VOS_NT_WINDOWS32` or `VOS__WINDOWS32` or `VOS_NT`? I'm not sure it matters much but the MSDN example uses `VOS__WINDOWS32`: https://learn.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource#examples

Reply:
yeah, I'm not sure it truly matters.  I'm copying what the `winresource` crate [is doing](https://github.com/BenjaminRi/winresource/blob/4435b7b82a638624045b7bacad55a97b84352639/lib.rs#L195)

[Elsewhere](https://android.googlesource.com/platform/external/pthreads/+/216eb8153b2455d1e15f8fbf84d4413ddcc40adc/version.rc)  I found:

```c
 VOS_NT_WINDOWS32 	File was designed for Windows Server 2003 family,
			Windows XP, Windows 2000, or Windows NT.
```

which suggests to me that `VOS__WINDOWS32` would be less correct (as it implies non-NT win32 like Windows 95 would be supported).  Not sure what just `VOS_NT` would mean.